python/pyspark/sql/connect/_typing.py

```diff
@@ -57,7 +57,7 @@ class UserDefinedFunctionLike(Protocol):
     deterministic: bool
 
     @property
-    def returnType(self) -> DataType:
+    def returnType(self) -> DataTypeOrString:
```

Comment:
do this introduce a behavior change?

https://github.com/apache/spark/blob/6ff760d483124b121d79c3a2d5fdc3ee3f27dd00/python/pyspark/sql/_typing.pyi#L70-L77

It seems that the PySpark UDF's `returnType` is always a `DataType`

cc @xinrong-meng

Reply:
Oh yes, I commented separately below.

Reply:
Yes, that's true.
However, the client is not available here and we don't have a proper way to parse the string here.

Reply:
Yeah, but I am still confused about it:

that the old implementation
`PySparkSession.builder.getOrCreate().createDataFrame(data=[], schema=data_type).schema` works.

I also tried 
```
    session = PySparkSession.builder.getOrCreate()
    parsed = session.client._analyze(  # type: ignore[attr-defined]
        method="ddl_parse", ddl_string=data_type
    ).parsed
```
and at least the tests passed.

But if I try
```
     parsed = PySparkSession.builder.getOrCreate().client._analyze(  # type: ignore[attr-defined]
        method="ddl_parse", ddl_string=data_type
    ).parsed
```
the tests always fail with `ValueError: Cannot invoke RPC on closed channel!`

Maybe we will have to add a pure python ddl parser, i don't know